mmdet/models/utils/transformer.py

```diff
@@ -1007,6 +1007,10 @@ def forward(self,
                     self.decoder.num_layers](output_memory) + output_proposals
 
             topk = self.two_stage_num_proposals
+            # We only use the first class in enc_outputs_class as foreground,
```

Comment:
We only use the first channel in enc_outputs_class as foreground,
 the other (num_classes - 1) channels are actually not used.
Its targets are set to be 0s, which indicates the first class (foreground) because we use [0, num_classes - 1] to indicate class labels, background class is indicated by num_classes (similar convention in RPN).
This follows the official implementation of Deformable DETR.